packages/core/src/util.test.ts

```diff
@@ -165,3 +170,93 @@ describe('getWorkflowRunStreamId', () => {
     expect(result.includes('_user')).toBe(true);
   });
 });
+
+describe('getPort', () => {
+  it('should return undefined or a positive number', async () => {
+    const port = await getPort();
+    expect(port === undefined || typeof port === 'number').toBe(true);
+    if (port !== undefined) {
+      expect(port).toBeGreaterThan(0);
+    }
+  });
+
+  it('should return a port number when a server is listening', async () => {
+    const server = http.createServer();
+
+    await new Promise<void>((resolve) => {
+      server.listen(0, () => resolve());
+    });
+
+    // Give system time to register the port
+    await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
Were you seeing failures without this? I've never seen `server.address()` fail after the listen callback is invoked (but maybe the mechanism that pid-port is using need this?)